File: frontend/src/constants/roles.js
/**
 * src/constants/roles.js
 * -----------------------
 * Single source of truth for role IDs on the frontend.
 *
 * These values MUST match the seeded role IDs in the database:
 *   1  = Super Admin
 *   2  = Hospital Admin
 *   3  = Receptionist
 *   4  = Doctor
 *   5  = Nurse
 *   6  = Lab Technician
 *   7  = Radiologist
 *   8  = Pharmacist
 *   9  = Billing Clerk
 *   10 = Inventory Manager
 *
 * DO NOT change these values.
 */

export const ROLES = {
  SUPER_ADMIN:       1,
  HOSPITAL_ADMIN:    2,
  RECEPTIONIST:      3,
  DOCTOR:            4,
  NURSE:             5,
  LAB_TECHNICIAN:    6,
  RADIOLOGIST:       7,
  PHARMACIST:        8,
  BILLING_CLERK:     9,
  INVENTORY_MANAGER: 10,
};

/**
 * Human-readable label for each role ID.
 * Used in headers, profile panels, and welcome banners.
 */
export const ROLE_LABELS = {
  [ROLES.SUPER_ADMIN]:       'Super Admin',
  [ROLES.HOSPITAL_ADMIN]:    'Hospital Admin',
  [ROLES.RECEPTIONIST]:      'Receptionist',
  [ROLES.DOCTOR]:            'Doctor',
  [ROLES.NURSE]:             'Nurse',
  [ROLES.LAB_TECHNICIAN]:    'Lab Technician',
  [ROLES.RADIOLOGIST]:       'Radiologist',
  [ROLES.PHARMACIST]:        'Pharmacist',
  [ROLES.BILLING_CLERK]:     'Billing Clerk',
  [ROLES.INVENTORY_MANAGER]: 'Inventory Manager',
};

File: app/core/constants/roles.py
"""
app/core/constants/roles.py
----------------------------
Role ID constants for the MediCore HMS.

These constants are the single source of truth for role IDs.
They replace all hardcoded integer literals throughout the codebase.

DO NOT change these values — they must match the seeded role IDs in main.py
and the actual rows in the roles table.

Role IDs in database:
  1  = Super Admin
  2  = Hospital Admin
  3  = Receptionist
  4  = Doctor
  5  = Nurse
  6  = Lab Technician
  7  = Radiologist
  8  = Pharmacist
  9  = Billing Clerk
  10 = Inventory Manager
"""


class RoleId:
    """Immutable role ID constants. Never use raw integers — use these."""

    SUPER_ADMIN:        int = 1
    HOSPITAL_ADMIN:     int = 2
    RECEPTIONIST:       int = 3
    DOCTOR:             int = 4
    NURSE:              int = 5
    LAB_TECHNICIAN:     int = 6
    RADIOLOGIST:        int = 7
    PHARMACIST:         int = 8
    BILLING_CLERK:      int = 9
    INVENTORY_MANAGER:  int = 10

    # Convenience sets for multi-role checks
    ADMIN_ROLES:    frozenset = frozenset({SUPER_ADMIN, HOSPITAL_ADMIN})
    CLINICAL_ROLES: frozenset = frozenset({DOCTOR, NURSE, LAB_TECHNICIAN, RADIOLOGIST})
    ALL_ROLES:      frozenset = frozenset(range(1, 11))
